### ISSUE TITLE
make 6.0.0 release

### ISSUE THREAD

**Issue by alifeee** [burnash/gspread#1359]
we should

- look through the changes, and make sure the readme migration guide is up to date
- anything else you can think of?

we should do this with plenty of time before xmas so that we can respond to (inevitable) incoming issues.

**Comment by lavigne958** [comment]
I think that's all, just check the readme is up to date then we can merge the branch and master together. The. Release

**Comment by NickCrews** [comment]
See https://github.com/burnash/gspread/issues/1360, I'm curious if you want to tweak the API of `Worksheet.update()` before you make this release. Thank you!

**Comment by alifeee** [comment]
> See https://github.com/burnash/gspread/issues/1360, I'm curious if you want to tweak the API of Worksheet.update() before you make this release. Thank you for all your effort in maintaining this lib!

We may change the warnings emitted in previous versions, but we will not tweak the APIs as that would be against the ideals of [semantic versioning](https://semver.org/)

**Comment by NickCrews** [comment]
Ah yes, I meant adjust the the 6.x API (and adjust the 5.x warnings to be in sync), but not change the 5.x API.

But it's a moot point, please ignore my previous comment. The 6.x API actually looks great, I just was reading the code on `master` instead of on the `feature/release_6_0_0` branch and was therefore confused :).

**Comment by lavigne958** [comment]
no problem , thank you for coming and giving us feedback that's gold for us.
If the warning is fine then we'll keep going with our schedule.

**Comment by TheOriginalSoni** [comment]
Out of curiosity, is there a planned date by when gspread 6.0 will be out? We're currently refactoring some of our bots, and curious when the update is supposed to be out at large

**Comment by lavigne958** [comment]
Hello, we can't provide a date as of yet. 

We don't have a specific date but we should be able to release it after the issue #1367 is solved.

I am currently very busy at work and can hardly find the time to review @alifeee 's PR.

I hope this week we can merge his first part.
Next week may be I can do my part. 

As of now I would say at least a few weeks.

**Comment by TheOriginalSoni** [comment]
Thank you for the information. And thanks for keeping this library updated, this has been a massive help!

**Comment by lavigne958** [comment]
We need to wait for #1380  to be done

Then we can release the new version.

**Comment by alifeee** [comment]
I have approved #1380 

As for the release notes concern, this is what the release notes look like:

<details><summary>show v6 release notes</summary>

## What's Changed
* Remove deprecated method delete_row by @cgkoutzigiannis in https://github.com/burnash/gspread/pull/1062
* Initial typing in client.py by @OskarBrzeski in https://github.com/burnash/gspread/pull/1159
* Split client http client by @lavigne958 in https://github.com/burnash/gspread/pull/1190
* Spelling fix & update docs with date_time_render_option behaviour by @alifeee in https://github.com/burnash/gspread/pull/1187
* #966  Add sketch typing for utils.py by @butvinm in https://github.com/burnash/gspread/pull/1196
* Remove accepted_kwargs decorator by @lavigne958 in https://github.com/burnash/gspread/pull/1229
* Remove/python-3.7 by @alifeee in https://github.com/burnash/gspread/pull/1234
* Bump isort from 5.11.4 to 5.12.0 by @dependabot in https://github.com/burnash/gspread/pull/1165
* bump flake8 to 6.0.0 by @alifeee in https://github.com/burnash/gspread/pull/1236
* merge master into 6.0.0 by @lavigne958 in https://github.com/burnash/gspread/pull/1241
* Remplace named tuples with enums by @lavigne958 in https://github.com/burnash/gspread/pull/1250
* Feature/add type hints worksheets by @lavigne958 in https://github.com/burnash/gspread/pull/1254
* Implement hex color conversion by @idonec in https://github.com/burnash/gspread/pull/1270
* remove lastUpdateTime by @alifeee in https://github.com/burnash/gspread/pull/1295
* Merge `master` into `feature/release_6_0_0` by @alifeee in https://github.com/burnash/gspread/pull/1320
* Add type checking to lint by @alifeee in https://github.com/burnash/gspread/pull/1337
* Warning/update swapped args by @alifeee in https://github.com/burnash/gspread/pull/1336
* Improve `Worksheet.sort()` signature by @lavigne958 in https://github.com/burnash/gspread/pull/1342
* Make `get_values` and alias of `get` by @alifeee in https://github.com/burnash/gspread/pull/1296
* fix type issue (remove `.first()` function) by @alifeee in https://github.com/burnash/gspread/pull/1344
* Remove/get records   use index by @alifeee in https://github.com/burnash/gspread/pull/1345
* increase warning stacklevel from 1 to 2 by @alifeee in https://github.com/burnash/gspread/pull/1361
* Feature/merge master by @lavigne958 in https://github.com/burnash/gspread/pull/1371
* feature/merge master by @lavigne958 in https://github.com/burnash/gspread/pull/1372
* Simplify get records by @alifeee in https://github.com/burnash/gspread/pull/1374
* Add util function `to_records` to build records by @lavigne958 in https://github.com/burnash/gspread/pull/1377
* feature/add utils get records by @lavigne958 in https://github.com/burnash/gspread/pull/1378
* Add migration guide for get_all_records by @lavigne958 in https://github.com/burnash/gspread/pull/1379

## New Contributors
* @cgkoutzigiannis made their first contribution in https://github.com/burnash/gspread/pull/1062
* @OskarBrzeski made their first contribution in https://github.com/burnash/gspread/pull/1159
* @butvinm made their first contribution in https://github.com/burnash/gspread/pull/1196
* @idonec made their first contribution in https://github.com/burnash/gspread/pull/1270

**Full Changelog**: https://github.com/burnash/gspread/compare/v5.12.4...v6.0.0

</details>

I think that is fine to put in a release. I think that we can also put something at the top linking to the migration guide, and listing the biggest changes (typing, ...?)

**Comment by alifeee** [comment]
after #1380 is complete I will make release 6.0.0 on Sunday at 12:00pm UTC

**Comment by lavigne958** [comment]
> I have approved #1380 
> 
> As for the release notes concern, this is what the release notes look like:
> 
> <details><summary>show v6 release notes</summary>
> 
> ## What's Changed
> * Remove deprecated method delete_row by @cgkoutzigiannis in https://github.com/burnash/gspread/pull/1062
> * Initial typing in client.py by @OskarBrzeski in https://github.com/burnash/gspread/pull/1159
> * Split client http client by @lavigne958 in https://github.com/burnash/gspread/pull/1190
> * Spelling fix & update docs with date_time_render_option behaviour by @alifeee in https://github.com/burnash/gspread/pull/1187
> * #966  Add sketch typing for utils.py by @butvinm in https://github.com/burnash/gspread/pull/1196
> * Remove accepted_kwargs decorator by @lavigne958 in https://github.com/burnash/gspread/pull/1229
> * Remove/python-3.7 by @alifeee in https://github.com/burnash/gspread/pull/1234
> * Bump isort from 5.11.4 to 5.12.0 by @dependabot in https://github.com/burnash/gspread/pull/1165
> * bump flake8 to 6.0.0 by @alifeee in https://github.com/burnash/gspread/pull/1236
> * merge master into 6.0.0 by @lavigne958 in https://github.com/burnash/gspread/pull/1241
> * Remplace named tuples with enums by @lavigne958 in https://github.com/burnash/gspread/pull/1250
> * Feature/add type hints worksheets by @lavigne958 in https://github.com/burnash/gspread/pull/1254
> * Implement hex color conversion by @idonec in https://github.com/burnash/gspread/pull/1270
> * remove lastUpdateTime by @alifeee in https://github.com/burnash/gspread/pull/1295
> * Merge `master` into `feature/release_6_0_0` by @alifeee in https://github.com/burnash/gspread/pull/1320
> * Add type checking to lint by @alifeee in https://github.com/burnash/gspread/pull/1337
> * Warning/update swapped args by @alifeee in https://github.com/burnash/gspread/pull/1336
> * Improve `Worksheet.sort()` signature by @lavigne958 in https://github.com/burnash/gspread/pull/1342
> * Make `get_values` and alias of `get` by @alifeee in https://github.com/burnash/gspread/pull/1296
> * fix type issue (remove `.first()` function) by @alifeee in https://github.com/burnash/gspread/pull/1344
> * Remove/get records   use index by @alifeee in https://github.com/burnash/gspread/pull/1345
> * increase warning stacklevel from 1 to 2 by @alifeee in https://github.com/burnash/gspread/pull/1361
> * Feature/merge master by @lavigne958 in https://github.com/burnash/gspread/pull/1371
> * feature/merge master by @lavigne958 in https://github.com/burnash/gspread/pull/1372
> * Simplify get records by @alifeee in https://github.com/burnash/gspread/pull/1374
> * Add util function `to_records` to build records by @lavigne958 in https://github.com/burnash/gspread/pull/1377
> * feature/add utils get records by @lavigne958 in https://github.com/burnash/gspread/pull/1378
> * Add migration guide for get_all_records by @lavigne958 in https://github.com/burnash/gspread/pull/1379
> 
> ## New Contributors
> * @cgkoutzigiannis made their first contribution in https://github.com/burnash/gspread/pull/1062
> * @OskarBrzeski made their first contribution in https://github.com/burnash/gspread/pull/1159
> * @butvinm made their first contribution in https://github.com/burnash/gspread/pull/1196
> * @idonec made their first contribution in https://github.com/burnash/gspread/pull/1270
> 
> **Full Changelog**: https://github.com/burnash/gspread/compare/v5.12.4...v6.0.0
> 
> </details>
> 
> I think that is fine to put in a release. I think that we can also put something at the top linking to the migration guide, and listing the biggest changes (typing, ...?)

It is ! It's actually smaller than I thought.

Yes a small line mentioning the migration guide is a good extra at the top of the release note. 

Yes we can add that too, we have: python version and typing so far.

**Comment by alifeee** [comment]
the release is made: <https://github.com/burnash/gspread/releases/tag/v6.0.0>

at some point in future we should remove the migration guide from the readme? in a few months/a year?

feel free to change the release decription to be more descriptive, especially about the new client.

now we wait...

**Comment by lavigne958** [comment]
> the release is made: https://github.com/burnash/gspread/releases/tag/v6.0.0
> 
> at some point in future we should remove the migration guide from the readme? in a few months/a year?
I think we can remove it when make the next major release.
Once 7.0.0 is out. No need for migration guide from 5 to 6.
At that time we can move it the documentation for the record, do anyone waking up late can still find it 🤔
> 
> feel free to change the release decription to be more descriptive, especially about the new client.
Yeah I saw the extra description, I realized we did not take the proper time to build it. I'll check if anything is worth mentioning, like you said may be mention what the new client does ? I don't know 🙄

> 
> now we wait...

Ladies and gentlemen brace for impact 😬
